### PR TITLE
Data transformation using view definition

### LINKED ISSUE #19: Implement KMP Data Transformer for View-Only UI Model Flattening
**Is your feature request related to a problem? Please describe.**

Working with raw FHIR resources at the UI level is cumbersome. When fetching data from the FHIR Engine (for example, a Patient with related Immunizations and Observations), the response is deeply nested and requires traversing included and revIncluded resources.

This creates a few problems:

* The UI becomes tightly coupled to FHIR structures
* Presentation code gets bloated with mapping logic
* Even simple screens (e.g., patient cards or lists) become unnecessarily complex

**Describe the solution you'd like**

Introduce a Data Transformer in the OHS Player KMP library.

This component should act as a **stateless, read-only layer** that:

* Takes raw Search API results from the FHIR Engine
* Transforms them into flat, lightweight, platform-agnostic Kotlin data classes (UI models)
* Performs the transformation on-the-fly, keeping the UI layer clean and focused

**Acceptance Criteria**

* Define a generic KMP interface (e.g., `FhirDataTransformer<T, R>`)
* Provide utilities for safely extracting `included` and `revIncluded` resources from search results
* Ensure transformations run on background threads (`Dispatchers.IO`) to avoid UI performance issues
* Keep the transformer strictly **read-only**; all write/CRUD operations should bypass it and interact directly with the FHIR Engine

**Describe alternatives you've considered**

**Persisting Flattened Models**
Storing already-flattened models in the database was also considered, but rejected because:

* Room cannot create tables dynamically. Room requires entities, DAOs, and the database class to be declared statically at compile time. ViewDefinition-driven table creation at runtime bypasses this entirely, meaning we would lose all Room abstractions and have to manage raw SQLite operations ourselves.
* DB sync complexity. Any write operation — whether user-initiated or triggered by a background sync event — would need to be reflected in both the FHIR Engine DB and the flat cache. Keeping these consistent, especially across sync boundaries, introduces a failure surface that is difficult to test and reason about.
* Code overhead. Implementing a custom search API on top of hand-managed flat tables duplicates functionality already provided by the FHIR Engine. This adds ongoing implementation, testing, and documentation burden with no acceptable trade-off given the library's current scope.

Note: Implementers who need flat table structures for their specific use case can build on top of the transformation layer this library provides. It is also a viable future iteration once the transformer API has stabilised.

**Additional context**

The transformer should integrate cleanly with the structure returned by the FHIR Engine search extensions, particularly how relationships are exposed (see [Search](https://github.com/ellykits/android-fhir/blob/master/engine/src/main/java/com/google/android/fhir/search/Search.kt) and [MoreSearch.kt](https://github.com/ellykits/android-fhir/blob/master/engine/src/main/java/com/google/android/fhir/search/MoreSearch.kt) in the Kotlin FHIR engine).

For large datasets:

* Consider using **Kotlin Sequences or Flow** to reduce memory overhead during transformations

**Would you like to work on the issue?**

Unassigned — open for contribution.

### PR DESCRIPTION
**Pull request by davidnjau**
Fixes #19 

This introduces the ViewDefinition model into ohs-player-library and makes it the sole driver of FHIR data extraction.
                                                                                                                                                                                                                      
The transformer now has a clear contract — it receives a resource and a ViewDefinition, evaluates the declared paths, and returns a typed state object. It does not care what the application wants or how it uses  the result.

The PR also introduces Data Transformations tests for the library.

### CODE REVIEW

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package dev.ohs.player.library.model
+
+import kotlinx.serialization.Serializable
+
+// TODO: Generate this class from StructureDefinition using tools like:
+//  https://github.com/hapifhir/org.hl7.fhir.core/tree/master/org.hl7.fhir.core.generator
+@Serializable
+data class ViewDefinition(
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt:27]: create `domain` package inside library then move the `model` and `transformer` packages inside it. i.e. `/domain/model` and `/domain/transformer`

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+@Serializable
+data class Constant(
+    val name: String? = null,
+    val valueBase64Binary: String? = null,
+    val valueBoolean: Boolean? = null,
+    val valueCanonical: String? = null,
+    val valueCode: String? = null,
+    val valueDate: String? = null,
+    val valueDateTime: String? = null,
+    val valueDecimal: String? = null,
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt]: Can we consider using Kotlin equivalent data types for these and any other types declared here?

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Transforms a FHIR resource into a generated Kotlin type T.
+ * Uses the OHS Kotlin FHIR SDK's FhirPathEngine for evaluation.
+ */
+class DataTransformer(
+    private val fhirPathEngine: FhirPathEngine,
+    @PublishedApi
+    internal val jsonConfiguration: Json = Json {
+        ignoreUnknownKeys = true
+        coerceInputValues = true
+    }
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: Extract this to JsonUtil object/singleton class for reuse. Add it to `util` package

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        val values = viewDefinition.allColumns()
+            .filter { it.name != null && it.path != null }
+            .associate { column ->
+                column.name!! to evaluatePath(resource, column.path!!)
+            }
+        return FhirExtractionResult(
+            resource = resource,
+            values = values,
+            json = jsonConfiguration)
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: With the Json provided as singleton class/ utility kotlin object you wouldn't need to have this constructor param, instead you would use it directly where this property (`jsonConfiguration`) is called.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    }
+
+    inline fun <reified T: Any> transform(resource: Resource, viewDefinition: ViewDefinition) : T =
+        extract(resource, viewDefinition).decode()
+
+    private fun evaluatePath(focus: Any?, path: String): String? {
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: Return `Any` to support all types instead of restricting the return type to a String.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        return try {
+            val results : Collection<Any> = fhirPathEngine.evaluateExpression(
+                expression = path,
+                base = focus,
+                variables = emptyMap()
+            )
+            results.firstOrNull()?.toString()
+        }catch (e: Exception){
+            null
+        }
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: Configure spotless then run `spotlessApply` to fix code formatting here.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    val valueId: String? = null,
+    val valueInstant: String? = null,
+    val valueInteger: Int? = null,
+    val valueInteger64: Long? = null,
+    val valueOid: String? = null,
+    val valueString: String? = null,
+    val valuePositiveInt: Int? = null,
+    val valueTime: String? = null,
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt]: Consider using kotlin equivalent for Time

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt (hunk 5 above)
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]:
```suggestion
    private fun evaluatePath(base: Any?, expression: String): Any? {
```

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    @PublishedApi
+    internal val jsonConfiguration: Json = Json {
+        ignoreUnknownKeys = true
+        coerceInputValues = true
+    }
+) : FhirExtractorRepository {
+    override fun extract(
+        resource: Resource,
+        viewDefinition: ViewDefinition
+    ): FhirExtractionResult {
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]:
```suggestion
    fun extract(
        resource: Resource,
        viewDefinition: ViewDefinition,
        contextVariablesMap: Map<String, Any> = emptyMap()
    ): FhirExtractionResult {
```

FHIRPath supports working with context variable which may be useful somehow during extraction we can provide it.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        extract(resource, viewDefinition).decode()
+
+    private fun evaluatePath(focus: Any?, path: String): String? {
+        return try {
+            val results : Collection<Any> = fhirPathEngine.evaluateExpression(
+                expression = path,
+                base = focus,
+                variables = emptyMap()
+            )
+            results.firstOrNull()?.toString()
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]:
```suggestion
            results.firstOrNull()
```

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractorRepository.kt [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package dev.ohs.player.library.transform
+
+import dev.ohs.fhir.model.r4.Resource
+import dev.ohs.player.library.model.ViewDefinition
+
+interface FhirExtractorRepository {
+    fun extract(resource: Resource, viewDefinition: ViewDefinition): FhirExtractionResult
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractorRepository.kt]: Delete this unnecessary abstraction, we do not expect to have multiple transformer classes. To simplify implementation and usage, we will stick to using `ViewDefinition`.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractorRepository.kt [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+class ExtractFhirDataUseCase(private val repository: FhirExtractorRepository){
+    operator fun invoke(resource: Resource, viewDefinition: ViewDefinition) : FhirExtractionResult =
+        repository.extract(resource, viewDefinition)
+}
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractorRepository.kt]: This class is unnecessary. Delete it.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractionResult.kt [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    val resource: Resource,
+    val values: Map<String, String?>,
+    @PublishedApi internal val json: Json = Json {
+        ignoreUnknownKeys = true
+        coerceInputValues = true
+    }
+) {
+    operator fun get(field: String): String? = values[field]
+
+    fun toJsonObject(): JsonObject = JsonObject(
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractionResult.kt]: I'd prefer returning a typed result instead of a JsonObject. That would render this class and the methods there in useless. Again there is no benefit constructing this class simply for transforming the result of the transformed data.

Move the implementation part of the `decode` method to   `DataTransformer#transform` method. 

If your  `DataTransformer#transform` returned a `Map<String, Any?>` you could use Kotlin collection [mapValues](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/map-values.html) to convert the values of the map from `Any?` to `JsonPrimitive` or `JsonNull` (if the value is null); `JsonNull` and `JsonPrimitive` are subclasses of [JsonElement](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-element/index.html)). 

Constructing the `JsonObject` would be as simple as passing the `mapOf<String, JsonElement>` as a constructor param for the [JsonObject](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-object/)

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt (hunk 6 above)
Comment (ellykits) [reply ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: You could refactor the `try-catch` block to Kotlin [runCatching](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/run-catching.html) method which handles exception just like `try-catch `and you can get the result or null. It's easier to read and works similar. Unless you intend to handle specific exception with try-catch block which seems not to be the case here.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractionResult.kt [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+        values.mapValues { (_, v) ->
+            if (v != null) JsonPrimitive(v) else JsonNull
+        }
+    )
+
+    inline fun <reified T : Any> decode(): T =
+        json.decodeFromJsonElement(toJsonObject())
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractionResult.kt]: Move the convertion of the JsonObject to Kotlin Model class to `DataTransformer#transform` function.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+            .associate { column ->
+                column.name!! to evaluatePath(resource, column.path!!)
+            }
+        return FhirExtractionResult(
+            resource = resource,
+            values = values,
+            json = jsonConfiguration)
+
+    }
+
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: Add Kotlin `kdoc` documentation for this function.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/transformer/DataTransformer.kt [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+import dev.ohs.fhir.fhirpath.FhirPathEngine
+import dev.ohs.fhir.model.r4.Resource
+import dev.ohs.player.library.model.ViewDefinition
+import kotlinx.serialization.json.Json
+
+
+/**
+ * Transforms a FHIR resource into a generated Kotlin type T.
+ * Uses the OHS Kotlin FHIR SDK's FhirPathEngine for evaluation.
+ */
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/transformer/DataTransformer.kt:51]: Improve and beef up this code documentation if possible provide a simple example for the code usage.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractionResult.kt [hunk 16]
```diff
@@ -0,0 +1,28 @@
+package dev.ohs.player.library.transform
+
+import dev.ohs.fhir.model.r4.Resource
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.decodeFromJsonElement
+
+class FhirExtractionResult(
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/FhirExtractionResult.kt]: See my comment on this class. You actually don't need this once you move the decode function logic to `DataTransformer#transform` method.

File: ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt [hunk 17]
```diff
@@ -0,0 +1,327 @@
...
+import kotlin.test.assertNull
+
+class DataTransformerTest {
+
+    private val fhirJson = FhirR4Json { ignoreUnknownKeys = true }
+    private val transformer = DataTransformer(FhirPathEngine.forR4())
+
+    // ── Shared resource ───────────────────────────────────────────────────────
+    // Extend this JSON to support new ViewDefinition expressions in future tests.
+
```
Comment (ellykits) [inline ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: Delete this

File: ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt [hunk 18]
```diff
@@ -0,0 +1,327 @@
...
+                select = listOf(SelectBlock(column = listOf(
+                    ViewColumn(name = "city",    path = "address.city.first()"),
+                    ViewColumn(name = "country", path = "address.country.first()")
+                )))
+            )
+        )
+        assertEquals("Nairobi", state.city)
+        assertEquals("KE",      state.country)
+    }
+
```
Comment (ellykits) [inline ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: These field level testing are really unnecessary. You can use one ViewDefinition to cover all these at one go.

Comment (ellykits) [reply ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: Delete these tests.

File: ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt [hunk 19]
```diff
@@ -0,0 +1,327 @@
...
+            )
+        )
+        assertEquals("P-001",                  state.patientId)
+        assertEquals("Smith John",             state.fullName)
+        assertEquals("+254700000001",          state.phone)
+        assertEquals("john.smith@example.com", state.email)
+        assertEquals("Nairobi",                state.city)
+    }
+
+    // ── Edge cases ────────────────────────────────────────────────────────────
```
Comment (ellykits) [inline ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: Delete this line an any line similar to this.

File: ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt [hunk 20]
```diff
@@ -0,0 +1,327 @@
...
+                    ViewColumn(name = "city",    path = "address.city.first()"),
+                    ViewColumn(name = "country", path = "address.country.first()")
+                )))
+            )
+        )
+        assertEquals("Nairobi", state.city)
+        assertEquals("KE",      state.country)
+    }
+
+    // ── Multiple fields in one ViewDefinition ─────────────────────────────────
```
Comment (ellykits) [inline ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: Delete

Comment (ellykits) [reply ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: You can just have one unit test for the happy path that uses a comprehensive `ViewDefinition` and run transform function against a resource that returns a State just like the function would be used in an app. You don't have to break them down into these parts. Instead you can test for edge cases.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt (hunk 5 above)
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]:
```suggestion
    private fun evaluatePath(focus: Any?, path: String): Any? {
```

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+                variables = emptyMap()
+            )
+            results.firstOrNull()?.toString()
+        }catch (e: Exception){
+            null
+        }
+    }
+
+    companion object{
+        fun forR4(): DataTransformer = DataTransformer(FhirPathEngine.forR4())
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: 👍

File: ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt [hunk 22]
```diff
@@ -0,0 +1,327 @@
...
+              }
+              """.trimIndent()
+        )
+    }
+
+    // ── Shared state class ────────────────────────────────────────────────────
+    // All fields are nullable with defaults. Each ViewDefinition populates only
+    // the fields it declares — unmapped fields remain null.
+    // Add new fields here when extending the patient JSON above.
+
```
Comment (ellykits) [inline ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: Convert this to kdoc documentation. Extract this state class to it's own file.

Comment (ellykits) [reply ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/DataTransformerTest.kt]: Can you fix the multiple issues you have with the test documentation in this file. Those multiple underscore on comments is really unconventional see below example and the likes. 

```kotlin
// ── Shared state class ────────────────────────────────────────────────────
```

File: ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/transformer/DataTransformerTest.kt [hunk 23]
```diff
@@ -0,0 +1,327 @@
...
+import dev.ohs.player.library.model.SelectBlock
+import dev.ohs.player.library.model.ViewColumn
+import dev.ohs.player.library.model.ViewDefinition
+import dev.ohs.player.library.transform.DataTransformer
+import kotlinx.serialization.Serializable
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+class DataTransformerTest {
```
Comment (ellykits) [inline ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/transformer/DataTransformerTest.kt:29]: Use the following [Kotlin method test naming convention](https://developer.android.com/kotlin/style-guide#function_names) for the test methods e.g. `transform_convertsResource_toState`

**Review by ellykits — CHANGES_REQUESTED** [review]
Thanks for the PR @davidnjau I have left a couple of comments. I know we have not integrated `spotlessApply`, it's done on another PR, you can integrate it fix code formatting issues which is yet to be merged.

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt (hunk 1 above)
Comment (davidnjau) [reply ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/model/ViewDefinition.kt:27]: Done

File: ohs-player-library/build.gradle.kts [hunk 24]
```diff
@@ -35,6 +54,12 @@ kotlin {
     }
 
     sourceSets {
+        commonMain.dependencies {
+            api(libs.kotlinx.serialization.json)
+            implementation("dev.ohs.fhir:fhir-model:1.0.0-beta03")
+            api("dev.ohs.fhir:fhir-path:1.0.0-beta02")
+            implementation("org.jetbrains.kotlinx:kotlinx-datetime:0.6.0")
```
Comment (ellykits) [inline ohs-player-library/build.gradle.kts]: Extract these to `.toml` file

File: ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/PatientState.kt [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Test state class representing extracted Patient fields.
+ *
+ * All fields are nullable with defaults. Each
+[ViewDefinition] populates only
+ * the fields it declares — unmapped fields remain null.
+ */
+@Serializable
+data class PatientState(
```
Comment (ellykits) [inline ohs-player-library/src/commonTest/kotlin/dev/ohs/player/library/PatientState.kt]: Move this to the test package

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+    /**
+     * Evaluates each FhirPath column declared in [viewDefinition] against [resource]
+     * and decodes the result into an instance of [T].
+     *
+     * @param resource the FHIR resource to extract data from
+     * @param viewDefinition declares the columns (name → FhirPath) to evaluate
+     * @param contextVariablesMap optional named variables available during FhirPath evaluation
+     * @return an instance of [T] with fields populated from the extracted values
+     */
+    inline fun <reified T : Any> transform(
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]:
```suggestion
    inline suspend fun <reified T : Any> transform(
```

Comment (ellykits) [reply ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]: To call the suspend function in test you can wrap it in:

```kotlin
 fun some_test() = runTest {
    val dataTransformer  = DataTransformer(...)
    dataTransformer.transform(...)
 } 
```

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+     *
+     * Columns with a null name or path are silently skipped.
+     *
+     * @param resource the FHIR resource to extract data from
+     * @param viewDefinition declares the columns to evaluate
+     * @param contextVariablesMap optional named variables for FhirPath evaluation
+     * @return a map of column name to extracted value; values are null if the
+     *         path yields no result or evaluation fails
+     */
+    fun extract(
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]:
```suggestion
    fun suspend fun extract(
```

File: ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+        contextVariablesMap: Map<String, Any> = emptyMap(),
+    ): Map<String, Any?> =
+        viewDefinition
+            .allColumns()
+            .filter { it.name != null && it.path != null }
+            .associate { column ->
+                column.name!! to evaluatePath(resource, column.path!!, contextVariablesMap)
+            }
+
+    private fun evaluatePath(
```
Comment (ellykits) [inline ohs-player-library/src/commonMain/kotlin/dev/ohs/player/library/domain/transform/DataTransformer.kt]:
```suggestion
    private suspend fun evaluatePath(
```

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — APPROVED** [review]
